evm/src/error.rs

```diff
@@ -0,0 +1,34 @@
+//! error handling and support
+
+use bytes::Bytes;
+use ethers::{abi::AbiEncode, utils::keccak256};
+use once_cell::sync::Lazy;
+use std::fmt::Display;
+
+// keccak(Error(string))
+pub static REVERT_PREFIX: [u8; 4] = [8, 195, 121, 160];
+
+/// Custom error prefiix
+pub static ERROR_PREFIX: Lazy<[u8; 32]> = Lazy::new(|| keccak256("CheatCodeError"));
+
+/// An extension trait for `std::error::Error` that can abi-encode itself
+pub trait SolError: std::error::Error {
```

Comment:
can we extend this to be a bit more advanced already now and just leave the more advanced stuff unused? e.g. as mentioned previously we probably want to be able to have a main error message, followed by an optional set of extra errors or help. take the `expectEmit` cheatcode for example: right now it doesn't tell you what didn't match in the logs when comparing, just that they didn't match. if we could display something like:

```
Error: Log didn't match expected log.
- Topic 0 matches
- Topic 1 matches
- Topic 2 doesn't match (a != b)
- ...
```

So essentially we'd want to revert with a custom Solidity error (not a string revert) that has the signature `error CheatCodeError(string err, string[] hints)` or something like that

the `SolError` trait could take care of the conversion so some errors just leave `hints` empty (or alternatively we can have two error types, one with hints and one without)